speech-commands/demo/index.js

```diff
@@ -324,3 +379,99 @@ startTransferLearnButton.addEventListener('click', async () => {
   startTransferLearnButton.textContent = 'Transfer learning complete.';
   startButton.disabled = false;
 });
+
+downloadFilesButton.addEventListener('click', () => {
+  const basename = getDatasetFileBasename();
+  const artifacts = transferRecognizer.serializeExamples();
+
+  // Trigger downloading of the manifest .json file.
+  const manifestAnchor = document.createElement('a');
+  manifestAnchor.download = `${basename}.json`;
+  manifestAnchor.href = window.URL.createObjectURL(
+      new Blob([artifacts.manifest], {type: 'application/json'}));
+  manifestAnchor.click();
+
+  // Trigger downloading of the data .bin file.
+  const weightDataAnchor = document.createElement('a');
+  weightDataAnchor.download = `${basename}.bin`;
+  weightDataAnchor.href = window.URL.createObjectURL(
+      new Blob([artifacts.data], {type: 'application/octet-stream'}));
+  weightDataAnchor.click();
+});
+
+/** Get the base name of the downloaded files based on current dataset. */
+function getDatasetFileBasename() {
+  if (transferRecognizer == null) {
+    throw new Error('Transfer model is unset.');
+  }
+  const items = [];
+  const exampleCounts = transferRecognizer.countExamples();
+  const labels = Object.keys(exampleCounts);
+  labels.sort();
+  for (const label of labels) {
+    const count = exampleCounts[label];
+    items.push(`${count}${label}`);
+  }
+  return items.join('-');
```

Comment:
This seems like it could hit filename length limits and imprison the user's data if they enter long-ish labels or many labels... i wonder if this should have a limit, or maybe just be a timestamp like YYYY-MM-DD-TTTT.json or something?